cypress/integration/spec.js

```diff
@@ -76,31 +88,90 @@ describe('Chart Explorer', function() {
     ).click()
     cy.get('[data-test=rm-encoding]').click()
 
-    // check for editor existence
-    cy.get('#configure-tabs-tab-editor').click()
-    cy.get('.monaco-editor')
-    cy.get('#configure-tabs-tab-builder').click()
-
     // check that titles work
     cy.get('[data-test=chart-title]').type('test title')
     cy.get('svg .role-title').contains('test title')
 
     // check that chart exists
     cy.get('[data-test=vega-embed]')
-
     cy.screenshot('page with viz')
 
-    // check download options
+    cy.get('[data-test=license-open]').click()
+    cy.get('[data-test=license-text]').contains('cypress license text')
+  })
+
+  it('Should be able to edit chart using editor', function() {
+    cy.get('#configure-tabs-tab-editor').click()
+
+    cy.get('.inputarea')
+      .type('{ctrl}f')
+      .focused()
+      .type('"title": "test 123",')
+
+    cy.get('#configure-tabs-tab-builder').click()
+
+    cy.get('svg .role-title').contains('test 123')
+  })
+
+  it('Should be able to download a chart', function() {
     cy.get('#dropdown-download').click()
 
+    // download as a vega-lite file format
     cy.get('.open > .dropdown-menu > :nth-child(3) > a').trigger('mousedown')
     cy.get('.open > .dropdown-menu > :nth-child(3) > a').should($m => {
       expect($m).to.have.length(1)
       expect($m.attr('href')).to.match(/^blob/)
     })
-    cy.get('#dropdown-download').click()
 
-    // check download as insight
+    // download as a vega file format
+    cy.get('.open > .dropdown-menu > :nth-child(2) > a').trigger('mousedown')
+    cy.get('.open > .dropdown-menu > :nth-child(2) > a').should($m => {
+      expect($m).to.have.length(1)
+      expect($m.attr('href')).to.match(/^blob/)
+    })
+
+    // download as a png file format
+    cy.get('.open > .dropdown-menu > :nth-child(5) > a').trigger('mousedown')
+    cy.get('.open > .dropdown-menu > :nth-child(5) > a').should($m => {
+      expect($m).to.have.length(1)
+      expect($m.attr('href')).to.match(/^data:image\/png/)
+    })
+
+    // download as a svg file format
+    cy.get('.open > .dropdown-menu > :nth-child(6) > a').trigger('mousedown')
+    cy.get('.open > .dropdown-menu > :nth-child(6) > a').should($m => {
+      expect($m).to.have.length(1)
+      expect($m.attr('href')).to.match(/^blob/)
+    })
+
+    // download as a HTML file format
+    cy.get('.open > .dropdown-menu > :nth-child(8) > a').trigger('mousedown')
+    cy.get('.open > .dropdown-menu > :nth-child(8) > a').should($m => {
+      expect($m).to.have.length(1)
+      expect($m.attr('href')).to.match(/^blob/)
+    })
+  })
+
+  it('Should be able to be shared', function() {
```

Comment:
I'm of the strong opinion that this should be pulled into a new file `integration/share.spec.js` instead of being another `it(...` step in a single test file. I would recommend creating a new `describe('Sharing'`) block and having separate `it('should share as an insight')`, etc.

Reply:
hmm I actually think it makes more sense to leave it this way then separate it into its own test file. Using `it` is one way of isolating tests and breaking up a large test suite into smaller isolated tests. And I think using a combination of `it`'s and new test files is the right thing to do when a test suite becomes large enough.

For these tests though, using just `it` is enough IMO. Everything being tested here lives on one page and the `it` tests are all logical (they relate to each other and don't test a completely unrelated component). I feel like splitting it up into a separate file is just over-engineering.

Reply:
I agree that leaving the "shared" test in here for now is ok, given that it's currently only one test. Sharing doesn't seem to bear any more weight than a concept like downloading or editing (which are other things being tested in this doc) enough to warrant its own file. If we someday build more tests specific to sharing, then maybe it would make sense to break it out.

Reply:
> Sharing doesn't seem to bear any more weight than a concept like downloading or editing

I was actually just trying to figure out what the other 3 tests were doing and adding a similar recommendation to split them into their own test files. I agree that if the test file is small, it's ok to generalize and keep various different tests together. But in this PR, we're adding tests that are specific to shared and expanding on download, so think they do deserve their own files so it's easier to later extend the tests instead of having to refactor them later.

Reply:
okay yeah, at that point we may as well have a separate `describe` for sharing, downloading, editing, etc, and if the length warrants it (which it probably will), breaking them into their own files feels ok to me.

Reply:
Are you suggesting to have 4 different files with an `it` block in each one? Or are you suggesting to have 1 file for the basic functionalities and then a second file for the remaining `it` tests?

Reply:
I spoke with Caroline and I think we'll be keeping it this way, but here's an article that pertains to this discussion that I found helpful: https://www.codesimplicity.com/post/what-is-overengineering/

Reply:
Ok, so it seems like we agree that continuing this style of adding onto a single test file is a bad idea in the long term, but for now just putting everything in here is sufficient since we're not sure about the future scope of chart-builder atm.

Reply:
@tkhanna42 yes. I think this is one of those cases where both options are equally "correct" given the state of the product. I think your intuition about wanting to break things up by context is good, and definitely something to keep up. But I don't think this codebase is yet at a point where it would introduce any real value. And as @DanialBetres pointed out, he's already "breaking up" the code by splitting up the original _one_ test into four `it` blocks. And based on how often we develop on chart-builder and the extent of its functionality, my senses tell me its okay to keep things together for now.